### PR TITLE
refactor(core): move React-injected services into ReactInjector directly

### PR DESCRIPTION
**Pull request by anotherchrisberry**
@jrsquared @christopherthielen PTAL

The `export let xxx = undefined;` approach is fine when importing directly from the declaring source file. However, if we attempt to re-export that field, e.g.

```javascript
// file: a/a.js
export let foo = undefined;
module('', []).service('foo', foo).run($injector => foo = $injector.get('foo'));

// file: a/index.js
export * from 'a.js'

// file: b.js
import { foo } from 'a';

foo === undefined // true
```

This becomes a problem when we try to package modules up as libraries, which we're trying to do.

Instead, we'll explicitly export the `ReactInjector`, and add those fields to it explicitly. I made them getters - not because of concern someone will overwrite them, but because we don't want to actually initialize something unless a component requests it; otherwise, tests get pretty gnarly.

Also changing the casing on `loadBalancerFilterModel`. I'll do the same with the other `*FilterModel`s in a separate PR.

### CODE REVIEW

**Comment by anotherchrisberry** [comment]
@danielpeach also PTAL - this adds a little complexity in tests, as we need to explicitly initialize the angular module containing the service.

File: app/scripts/modules/core/react/react.injector.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  public get pipelineConfigService() { return this.$injector.get('pipelineConfigService') as PipelineConfigService; }
+  public get manualJudgmentService() { return this.$injector.get('manualJudgmentService') as ManualJudgmentService; }
+  public get pipelineTemplateService() { return this.$injector.get('pipelineTemplateService') as PipelineTemplateService; }
+  public get variableInputService() { return this.$injector.get('variableInputService') as VariableInputService; }
+  public get variableValidatorService() { return this.$injector.get('variableValidatorService') as VariableValidatorService; }
+  public get schedulerFactory() { return this.$injector.get('schedulerFactory') as SchedulerFactory; }
+  public get infrastructureSearchService() { return this.$injector.get('infrastructureSearchService') as InfrastructureSearchService; }
+  public get waypointService() { return this.$injector.get('waypointService') as WaypointService; }
+
+  public give(fn: IWantInjector): void {
```
Comment (christopherthielen) [inline app/scripts/modules/core/react/react.injector.ts]: I suspect we don't need the `give` queue anymore, which also means we wouldn't need `flush` either

Comment (anotherchrisberry) [inline app/scripts/modules/core/react/react.injector.ts]: That's true, although you've reminded me why I need to deal with the `angular2react` components (didn't do that in this PR but will add them).

Still need the `flush` or something similar to get the `$injector` into the class, but will rename it to `initialize` or something.

**Comment by anotherchrisberry** [comment]
@christopherthielen put the `angular2react` bits in a separate commit - quite the injector we're building up here, but...it'll go away eventually?

File: app/scripts/modules/core/delivery/executionGroup/execution/Execution.tsx [hunk 2]
```diff
@@ -267,6 +262,7 @@ export class Execution extends React.Component<IExecutionProps, IExecutionState>
   }
 
   public render() {
+    const { AccountTag, ExecutionStatus, PipelineGraph, ExecutionDetails, CopyToClipboard } = ReactInjector;
```
Comment (christopherthielen) [inline app/scripts/modules/core/delivery/executionGroup/execution/Execution.tsx]: I think this reads a bit better in JSX than the other way `<ReactInjector.SomeComponent/>`

File: app/scripts/modules/core/loadBalancer/LoadBalancers.tsx [hunk 3]
```diff
@@ -5,11 +5,9 @@ import { Subscription } from 'rxjs';
 
 import { Application } from 'core/application/application.model';
 import { FilterTags, IFilterTag } from 'core/filterModel/FilterTags';
-import { HelpField } from 'core/help/HelpField';
 import { ILoadBalancer, ILoadBalancerGroup } from 'core/domain';
 import { LoadBalancerPod } from './LoadBalancerPod';
 import { StickyContainer } from 'core/utils/stickyHeader/StickyContainer';
-import { Spinner } from 'core/widgets/Spinner';
```
Comment (christopherthielen) [inline app/scripts/modules/core/loadBalancer/LoadBalancers.tsx]: I see one downside is that we lose this context (`core/widgets`)

**Comment by jrsquared** [comment]
Oh my. Not going to lie: I hate this. It is __super__ ugly. It was so nice to be able to just pretend things were actually ES6 modules we could import. I don't have a better suggestion though. Especially for the React component bit. Granted, in theory those should diminish quickly since that's mainly my fault for converting Executions right in the middle of angular world.

We basically just created our own `window`.

**Comment by christopherthielen** [comment]
> We basically just created our own window.

I see this as a working list that we will slowly chip away.

**Comment by jrsquared** [comment]
> I see this as a working list that we will slowly chip away.

And with it looking like this, I will s/slowly/quickly/ ;)

**Comment by christopherthielen** [comment]
Nice!  we have a volunteer

**Comment by anotherchrisberry** [comment]
Get to work, Justin

**Pull request merged**